src/sandbox/value_estimates/value_estimates.py

```diff
@@ -0,0 +1,27 @@
+from abc import ABC, abstractmethod
+from typing import Hashable
+
+from sandbox.algorithms.algorithm import Action, Observation
+
+
+class StateValueEstimates(ABC):
+    """An array with estimates of state-value function or action-value"""
+    @abstractmethod
+    def __getitem__(self, observation: Observation) -> float:
+        ...
+
+    @abstractmethod
+    def __setitem__(self, observation: Observation, value: float) -> None:
+        ...
+
+
+class ActionValueEstimates(ABC):
```

Comment:
As as I understand, this estimates could be represented as a table or as a neural network making state approximation.
But despite of underlying representation, it always should return a value for a given state (or state, action pair)

Reply:
In case of neural network, as forward pass will most likely return values for all of the actions, it will be very inefficient to call `__getitem__`  independently for every possible action.